src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/RetryPolicy.Generic.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+
+namespace Microsoft.Data.SqlClient.Reliability
+{
+    /// <summary>
+    /// Provides a generic version of the <see cref="RetryPolicy"/> class.
+    /// </summary>
+    /// <typeparam name="T">The type that implements the <see cref="ITransientErrorDetectionStrategy"/> interface that is responsible for detecting transient conditions.</typeparam>
+    public class RetryPolicy<T> : RetryPolicy where T : ITransientErrorDetectionStrategy, new()
```

Comment:
Can we rename this to be SqlRetryPolicy, and apply similar naming pattern for all the public classes to be inline with other classes that this library offers?